src/DistributedComputations/plan_distributed_transforms.jl

```diff
@@ -0,0 +1,47 @@
+using Oceananigans.Architectures: array_type
+using Oceananigans.Solvers: plan_forward_transform, plan_backward_transform, DiscreteTransform
+using Oceananigans.Solvers: Forward, Backward
+
+@inline reshaped_size(grid) = size(grid, 2), size(grid, 1), size(grid, 3)
+
+struct FFTPlan{F, B}
+     forward :: F
+    backward :: B
+end
```

Comment:
How is this specific to distributed transforms? Seems generic

Reply:
Actually, there is no need to have a new type. I have changed this to a named tuple

Reply:
Okay right I think that's what is used for the non-disttributed